x-pack/plugin/async-search/src/main/java/org/elasticsearch/xpack/search/TransportGetAsyncStatusAction.java

```diff
@@ -67,15 +71,44 @@ protected void doExecute(Task task, GetAsyncStatusRequest request, ActionListene
         AsyncExecutionId searchId = AsyncExecutionId.decode(request.getId());
         DiscoveryNode node = clusterService.state().nodes().get(searchId.getTaskId().getNodeId());
         DiscoveryNode localNode = clusterService.state().getNodes().getLocalNode();
+
         if (node == null || Objects.equals(node, localNode)) {
-            store.retrieveStatus(
-                request,
-                taskManager,
-                AsyncSearchTask.class,
-                AsyncSearchTask::getStatusResponse,
-                AsyncStatusResponse::getStatusFromStoredSearch,
-                listener
-            );
+            if (request.getKeepAlive() != null && request.getKeepAlive().getMillis() > 0) {
+                long expirationTime = System.currentTimeMillis() + request.getKeepAlive().getMillis();
+                store.updateExpirationTime(searchId.getDocId(), expirationTime, ActionListener.wrap(p -> {
```

Comment:
This new code is modeled after similar code in `AsyncResultsService` which is what the `GET _async_search/<id>` code path uses.

Reply:
Thanks. I am assuming that you took the existing code as-is without additional changes. Correct?

Reply:
As much as possible. There are some differences, but those were already there between the search and status endpoint.

The structure is:
if keep_alive present and > 0:
 -> store.updateExpirationTime(docId, expirationTime, ActionListener-for-result

That part is the same. In the ActionListener, the exception/failure handler is the same. But onResponse handler differs, but in the same way these two endpoints already did. The status endpoint just calls `store.retrieveStatus` whereas the search endpoint does additional steps. Both need to set `task.setExpirationTime(expirationTimeMillis);` on the AsyncSearchTask, so that is new to the status endpoint. The search endpoint does that down in the `AsyncResultsService.getSearchResponseFromTask` method.